v2/internal/reconcilers/arm/arm_client_cache.go

```diff
@@ -234,9 +230,12 @@ func (c *ARMClientCache) newCredentialFromSecret(secret *v1.Secret, nsName types
 		}
 	} else {
 		// Here we check for workload identity if client secret is not provided.
-		credential, err = newWorkloadIdentityCredential(string(tenantID), string(clientID), tokenFile)
+		credential, err = identity.NewWorkloadIdentityCredential(string(tenantID), string(clientID))
 		if err != nil {
-			err = errors.Wrapf(err, "credential secret %q does not contain key %q and failed to get workload identity credential for clientID %q from %q ", nsName.String(), config.AzureClientSecretVar, string(clientID), tokenFile)
+			err = errors.Wrapf(
+				err,
+				"credential secret %q does not contain key %q and failed to get workload identity credential for clientID %q from %q ",
+				nsName.String(), config.ClientSecretVar, string(clientID), identity.TokenFile)
```

Comment:
If you wrap some parameters onto separate lines, wrap them all.

```suggestion
				nsName.String(), 
				config.ClientSecretVar, 
				string(clientID), 
				identity.TokenFile)
```